modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RapidExperimentationTest.java

```diff
@@ -0,0 +1,328 @@
+        expList.add(rapidExperiment2);
+        expList.add(rapidExperiment3);
+        for (int i = 1; i <= 3; i++) {
+            Experiment exp = expList.get(i - 1);
+            List<Bucket> bucketList = BucketFactory.createBuckets(exp, 3);
+            postBuckets(bucketList);
+
+            exp.state = Constants.EXPERIMENT_STATE_RUNNING;
+            exp = putExperiment(exp);
+
+        }
+
+    }
+
+    @Test
+    public void testRapidExperiment() {
+
+        Experiment exp = getExperiment(expList.get(0));
+        for (int i = 1; i <= 5; i++) {
+            // lets do an assignment for a user with context set to PROD
+            Assignment assignment = postAssignment(exp, new User("user" + i), "PROD");
+
+            // lets assert the status and response code and also the state of the experiment
+            assertReturnCode(response, HttpStatus.SC_OK);
+            Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+            Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_RUNNING);
+        }
+
+        // since the experiment crossed the maxusers the experiment should change to PUASED
+        Assignment assignment = postAssignment(exp, new User("user10"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        assignment = postAssignment(exp, new User("user11"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        assignment = postAssignment(exp, new User("user12"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        Assert.assertEquals(assignment.status, "EXPERIMENT_PAUSED");
+        exp = getExperiment(exp);
+        Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_PAUSED);
+        expList.set(0, rapidExperiment1);
+    }
+
+    /**
+     * This test case covers a scenario where we start an experiment, do assignments and after that update the
+     * experiment to rapid experiment with max users less than the number of users already assigned
+     */
+    @Test
+    public void testChangeRegularExperimentToRapidExperiment() {
+        clearAssignmentsMetadataCache();
+        Experiment exp = getExperiment(expList.get(1));
+
+        for (int i = 1; i <= 5; i++) {
+            // lets do an assignment for a user with context set to PROD
+            Assignment assignment = postAssignment(exp, new User("user" + i), "PROD");
+
+            // lets assert the status and response code and also the state of the experiment
+            assertReturnCode(response, HttpStatus.SC_OK);
+            Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+            Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_RUNNING);
+        }
+
+        exp = exp.setIsRapidExperiment(true).setUserCap(3);
+        exp = putExperiment(exp);
+
+        clearAssignmentsMetadataCache();
```

Comment:
@sudheerkethireddy I don't think this will work when we run integration tests against a wasabi cluster instead of a single node. Can we put a note here or create a test profile where we disable this test when running against a cluster?